plugin/src/main/java/io/jenkins/plugins/casc/SecretSourceResolver.java

```diff
@@ -26,24 +26,23 @@
     private static final String escapeEnclosedBy = escapedWith + enclosedBy;
 
     private static final Logger LOGGER = Logger.getLogger(SecretSourceResolver.class.getName());
-    private static final StringSubstitutor NULL_SUBSTITUTOR = new StringSubstitutor(
-        UnresolvedLookup.INSTANCE);
-    private static final StringSubstitutor SUBSTITUTOR = new StringSubstitutor(
-        StringLookupFactory.INSTANCE.interpolatorStringLookup(
-            ImmutableMap.of(
-                "base64", StringLookupFactory.INSTANCE.base64EncoderStringLookup(),
-                "file", FileStringLookup.INSTANCE
-            ),
-            ConfigurationContextStringLookup.INSTANCE, false));
-
-    static {
-        SUBSTITUTOR
+    private final StringSubstitutor nullSubstitutor;
+    private final StringSubstitutor substitutor;
+
+    public SecretSourceResolver(ConfigurationContext configurationContext) {
```

Comment:
Made the class none static to reduce memory footprint of the `StringSubstitutor` when JCasC is actively using them running.